runtime/weak.c

```diff
@@ -105,7 +105,7 @@ CAMLexport value caml_ephemeron_create (mlsize_t len)
   for (i = 1; i < size; i++) Field (res, i) = caml_ephe_none;
   Field (res, CAML_EPHE_LINK_OFFSET) = caml_ephe_list_head;
   caml_ephe_list_head = res;
-  return caml_check_urgent_gc(res);
+  return res;
```

Comment:
Why removing this call?

Reply:
It seems to me that you made it a polling location for the purpose of running memprof callbacks, and it is a public function.

Reply:
That's also a matter of coherence with the corresponding `Array` function. In order to avoid doing this in the public function, the simplest is probably to move to call to `caml_weak_create`/`caml_ephe_create`. The same applies to `caml_ephe_get_data`, `caml_ephe_get_key` and `caml_ephe_get_key_copy`. And we can do poling in all these functions.

Reply:
Done for `caml_ephe_create`, `caml_ephe_get_data_copy`, and `caml_ephe_get_key_copy`. Why `caml_ephe_get_data` and `caml_ephe_get_key`?

Reply:
> Why `caml_ephe_get_data` and `caml_ephe_get_key`?

They both allocate an option from C code. So they don't do polling. When called from OCaml, they thus act as allocators and could do polling. That's really a nitpick , I have to admit!

Reply:
Nitpick or not, making me look at the code for optionalize made me realise that my implementation was wrong (C didn't stop the type wrongful conversions int -> value -> int). The int ended up registered as a root so this could cause a crash. Tests didn't catch that either.

Reply:
Done for optionalize as well.